File: Dart/asynchronous.dart
void main() {
  test();
}

Future<bool> login(String username, String password) async => username.isNotEmpty && password.isNotEmpty;

test() async {
  await login('turaiiao', '12345678').then((data) {
    if (data) {
      print('Login successfully !');
    }
  });
}

File: JavaScript/asynchronous.js
let login = async (username, password) => {
  return username != '' && password != ''
}

let test = async () => {
  await login('turaiiao', '12345678').then((data) => {
    if (data) {
      console.log('Login successfully !')
    }
  })
}

test(); 
